operator/api/v1alpha1/kyma_types.go

```diff
@@ -250,11 +259,19 @@ type KymaCondition struct {
 	// +optional
 	TemplateInfo TemplateInfo `json:"templateInfo,omitempty"`
 
+	// ModuleInfo provide the latest deployed Module CR information.
+	ModuleInfo ModuleInfo `json:"moduleInfo,omitempty"`
+
 	// Timestamp for when Kyma last transitioned from one status to another.
 	// +optional
 	LastTransitionTime *metav1.Time `json:"lastTransitionTime,omitempty"`
 }
 
+type ModuleInfo struct {
+	// Generation tracks the active Generation of the Deployed Module CR.
+	Generation int64 `json:"generation,omitempty"`
```

Comment:
Why would you need to track the generation of the module CR? Since you only want to react to state changes those would not invoke a generation increment, but a resource version increment instead

Reply:
removed